gradle-junit-reports/src/main/java/com/palantir/gradle/junit/JunitReportsFinalizer.java

```diff
@@ -24,34 +24,38 @@
 import javax.inject.Inject;
 import javax.xml.transform.TransformerException;
 import org.gradle.api.DefaultTask;
+import org.gradle.api.Project;
 import org.gradle.api.Task;
 import org.gradle.api.file.Directory;
 import org.gradle.api.file.DirectoryProperty;
 import org.gradle.api.file.RegularFileProperty;
 import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.util.GUtil;
 import org.w3c.dom.Document;
 
 public class JunitReportsFinalizer extends DefaultTask {
 
     public static void registerFinalizer(
-            Task task, TaskTimer timer, FailuresSupplier failuresSupplier, Provider<Directory> reportDir) {
-        JunitReportsFinalizer finalizer = Tasks.createTask(
-                task.getProject().getTasks(), task.getName() + "CircleFinalizer", JunitReportsFinalizer.class);
-        if (finalizer == null) {
-            // Already registered (happens if the user applies us to the root project and subprojects)
-            return;
-        }
-        finalizer.setStyleTask(task);
-        finalizer.setTaskTimer(timer);
-        finalizer.setFailuresSupplier(failuresSupplier);
-        finalizer
-                .getTargetFile()
-                .set(reportDir.map(dir -> dir.file(task.getProject().getName() + "-" + task.getName() + ".xml")));
-        finalizer.getReportDir().set(reportDir);
-
-        task.finalizedBy(finalizer);
+            Project project,
+            String taskName,
+            TaskTimer timer,
+            FailuresSupplier failuresSupplier,
+            Provider<Directory> reportDir) {
+        TaskProvider<JunitReportsFinalizer> finalizer = project.getTasks()
+                .register(GUtil.toLowerCamelCase(taskName + " CircleFinalizer"), JunitReportsFinalizer.class, task -> {
```

Comment:
It's probably fine to use this method, but Gradle regards these `G` prefixed classes as internal (despite not being in the `.internal` package - `GFileUtils` is another example) and regularly breaks/removes methods from them in point releases 😱